### PR TITLE
v4 migration guide

### PR DESCRIPTION
**Pull request by tristanlee85**

### CODE REVIEW

**Comment by tristanlee85** [comment]
@ierceg can you review this to make sure I didn't leave anything out in migrating from v3 to v4?

File: guides/layer0_migration.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Layer0 Version 2.x.x to 3.x.x Migration Guide
 
-This guide describes what you need to know when migrating from version 2.x.x to version 3.x.x. Note that you can continue to use version 2.x.x but we highly recommend upgrading to version 3.x.x. The biggest key change is that in keeping with industry best practices, v3 of the platform now requires Node 14 whereas v2 was on Node 12. 
+This guide describes what you need to know when migrating from version 2.x.x to version 3.x.x. Note that you can continue to use version 2.x.x but we highly recommend upgrading to version 3.x.x. The biggest key change is that in keeping with industry best practices, v3 of the platform now requires Node 14 whereas v2 was on Node 12.
```
Comment (ierceg) [inline guides/layer0_migration.md:3]: I think it's time to deprecate this guide. There should be just one guide: from v1/2/3 to v4. All other versions have been deprecated.

File: guides/layer0_migration_v4.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+You will need to upgrade to the latest Layer0 packages available on NPM. To upgrade your packages, you will need to install the latest version of the Layer0 CLI. You should install this globally using: `npm i -g @layer0/cli`.
+
+Once installed, change into your project’s directory. To install the latest Layer0 dependencies, run `layer0 use latest`. This will update `package.json` with the new Layer0 4.x dependencies and scripts for building and deploying your site.
+
+Build your project and verify everything is functioning as expected. You should then deploy your site to your testing environment and re-verify functionality.
+
+## Edge links will change automatically
+
+Once you deploy the new version of Layer0, your Edge links will change to have the format `*.layer0-limelight.link`. Your current edge links will switch to the new format on the next deployment for that environment. Old edge links will continue to work even after the first deployment. Permalinks will remain `*.free.layer0-perma.link`.
```
Comment (ierceg) [inline guides/layer0_migration_v4.md:23]:
```suggestion
Once you deploy the new version of Layer0, your Edge links will change to have the format `*.layer0-limelight.link`. Your current edge links will switch to the new format on the next deployment for that environment. Old edge links may continue to work correctly for a while but they are considered deprecated and will be turned off without further notice. Permalinks will remain `*.free.layer0-perma.link`.
```

File: guides/layer0_migration_v4.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+## DNS will need updating for custom domains
+
+If you have a custom domain pointing to Layer0, you will need to update your A/CNAME record. To obtain the new network details, navigate to https://app.layer0.co and go to your site. Click the _Environments_ tab and choose the environment that has the custom domain. From there, click on the _Networking_ tab.
+
+You will see 2 different DNS configurations. If you are using a sub-domain, you will need to update the CNAME record with the value provided. If you are using an apex domain, you will need to update the A records to match the Anycast IP addresses listed.
```
Comment (ierceg) [inline guides/layer0_migration_v4.md:29]:
```suggestion
You will see two different DNS configurations. If you are using a sub-domain, you will need to update the CNAME record with the value provided. If you are using an apex domain, you will need to update the A records to match the Anycast IP addresses listed.
```

**Review by ierceg — CHANGES_REQUESTED** [review]

File: guides/layer0_migration.md (hunk 1 above)
Comment (tristanlee85) [reply guides/layer0_migration.md:3]: Is it safe to assume that 1/2/3 > 4 is the same process? I can't recall much about 1.x.

Comment (ierceg) [reply guides/layer0_migration.md:3]: It is safe to assume, yes.

**Pull request merged**